test/http-request.ts

```diff
@@ -0,0 +1,332 @@
+        method: 'GET',
+      } as http.IncomingMessage;
+      const wrapper = makeHeaderWrapper(req as unknown as http.IncomingMessage);
+      assert.strictEqual(wrapper, null);
+    });
+  });
+
+  describe('getCloudTraceContext', () => {
+    it('should extract trace & span from X-Cloud-Trace-Context', () => {
+      const tests = [
+        {
+          header: '105445aa7843bc8bf206b120001000/000000001;o=1',
+          expected: {
+            trace: '105445aa7843bc8bf206b120001000',
+            spanId: '000000001',
+            traceSampled: true,
+          },
+        },
+        // TraceSampled is false
+        {
+          header: '105445aa7843bc8bf206b120001000/000000001;o=0',
+          expected: {
+            trace: '105445aa7843bc8bf206b120001000',
+            spanId: '000000001',
+            traceSampled: false,
+          },
+        },
+        {
+          // No span
+          header: '105445aa7843bc8bf206b120001000;o=1',
+          expected: {
+            trace: '105445aa7843bc8bf206b120001000',
+            spanId: undefined,
+            traceSampled: true,
+          },
+        },
+        {
+          // No trace
+          header: '/105445aa7843bc8bf206b120001000;o=0',
+          expected: {
+            trace: undefined,
+            spanId: '105445aa7843bc8bf206b120001000',
+            traceSampled: false,
+          },
+        },
+        {
+          // No traceSampled
+          header: '105445aa7843bc8bf206b120001000/0',
+          expected: {
+            trace: '105445aa7843bc8bf206b120001000',
+            spanId: '0',
+            traceSampled: false,
+          },
+        },
+        {
+          // No input
+          header: '',
+          expected: {
+            trace: undefined,
+            spanId: undefined,
+            traceSampled: false,
+          },
+        },
+      ];
+      for (const test of tests) {
+        const req = {
+          method: 'GET',
+        } as unknown as http.IncomingMessage;
+        req.headers = {
+          'x-cloud-trace-context': test.header,
+        };
+
+        const wrapper = makeHeaderWrapper(req);
+        const context = getCloudTraceContext(wrapper!);
+        if (context) {
+          assert.strictEqual(
+            context.trace,
+            test.expected.trace,
+            `From ${test.header}; Expected trace: ${test.expected.trace}; Got: ${context.trace}`
+          );
+          assert.strictEqual(
+            context.spanId,
+            test.expected.spanId,
+            `From ${test.header}; Expected spanId: ${test.expected.spanId}; Got: ${context.spanId}`
+          );
+          assert.strictEqual(
+            context.traceSampled,
+            test.expected.traceSampled,
+            `From ${test.header}; Expected traceSampled: ${test.expected.traceSampled}; Got: ${context.traceSampled}`
+          );
+        } else {
+          assert(false);
+        }
+      }
+    });
+  });
+
+  describe('getOrInjectTraceParent', () => {
```

Comment:
similarly, please be more descriptive about what is being tested in this series of tests? Method name is not enough - we should describe the use-case (e.g. w3 trace extraction and propagation)

Reply:
It's convention to describe `function name` and then be more descriptive within the it() statements. I'm going to pepper more comments throughout the tests, but am not keen to change the describe snippets. 

It's impt to keep the current readability: 

![Screen Shot 2021-06-09 at 10 45 53 PM](https://user-images.githubusercontent.com/69952136/121471394-73bbaa00-c974-11eb-89df-ac734ac1aa95.png)